pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -358,21 +358,21 @@ func (inClusterClientConfig) ClientConfig() (*restclient.Config, error) {
 	return restclient.InClusterConfig()
 }
 
-func (inClusterClientConfig) Namespace() (string, error) {
```

Comment:
> this function never actually returns an error. Remove the error parameter or use it instead of the bool.

It's implementing an interface.  Other implementors do return errors.